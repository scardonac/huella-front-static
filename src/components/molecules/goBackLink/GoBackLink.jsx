//Dependencies
import { useContext } from 'react'
//Context
import { NavigateAppContext } from '../../../context/NavigateAppContext'

export const GoBackLink = ({ left = "30px", top = "30px" }) => {
    const { goBack } = useContext(NavigateAppContext)
    return (
        <div
            id='go_back_div'
            data-testid="go_back_div"
            className='relative'
        >
            <a
                className='text-primary-gris3 underline absolute cursor-pointer'
                style={{ left: left, top: top }}
                onClick={goBack}
                id='go_back_link'
                data-testid="go_back_link"
            >
                {"< Volver"}
            </a>
        </div>
    )
}