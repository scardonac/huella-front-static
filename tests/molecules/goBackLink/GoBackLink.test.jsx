import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import { act } from 'react-dom/test-utils';
import { GoBackLink } from '../../../src/components/molecules/goBackLink/GoBackLink';
//mocks svg
jest.mock('../../../src/assets/icons/IconProvider', () => ({
    Icons: {
        IconAlertWhite: 'IconAlertWhite',
        AddDocumentBlackIcon: 'AddDocumentBlackIcon',
    }
}));
jest.mock('../../../src/assets/Illustrations/IllustrationProvider', () => ({
    Illustrations: {
        Add_OtherCategory_Azul: 'Add_OtherCategory_Azul',
    }
}));
//mocks context
jest.mock('../../../src/context/NavigateAppContext', () => ({
    NavigateAppContext: {
        goBack: jest.fn(),
    }
}));

describe('Pruebas en <GoBackLink />', () => {

    test('Debe renderizarse correctamente', () => {

        // Mock the useContext hook
        const mockContext = {
            goBack: jest.fn()
        };
        jest.spyOn(React, 'useContext').mockReturnValue(mockContext);

        const { baseElement } = render(<GoBackLink />);
        expect(baseElement).toMatchSnapshot();
    });

    test('Debe renderizarse correctamente con estilos', () => {
        render(<GoBackLink style="test" />);

        const fnLink = screen.getByTestId('go_back_link');

        act(() => {
            fireEvent.click(fnLink);
        });

    });

});