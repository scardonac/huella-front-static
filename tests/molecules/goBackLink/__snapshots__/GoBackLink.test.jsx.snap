// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GoBackLink /> Debe renderizarse correctamente 1`] = `
<body>
  <div>
    <div
      class="relative"
      data-testid="go_back_div"
      id="go_back_div"
    >
      <a
        class="text-primary-gris3 underline absolute cursor-pointer"
        data-testid="go_back_link"
        id="go_back_link"
        style="left: 30px; top: 30px;"
      >
        &lt; Volver
      </a>
    </div>
  </div>
</body>
`;
